backend/zerowaste/src/main/java/com/zerowaste/models/product/Product.java

```diff
@@ -187,6 +187,13 @@ public void setDeletedAt(LocalDate deletedAt) {
         this.deletedAt = deletedAt;
     }
 
+    public void setPromotions(Set<Promotion> promotions2) {
+        if (promotions2 == null) {
+            throw new IllegalArgumentException("Promoções não podem ser nulas.");
+        }
+        this.promotions = Collections.unmodifiableSet(promotions2);
+    }
+
```

Comment:
Aqui o caso do `setPromotion()` citado anteriormente. Deixe apenas assim, sem validações.

Depois faz o `getPromotion()`. Vê os outros getters que tem no arquivo e repete a mesma lógica.

```suggestion
   public void setPromotions(Set<Promotion> promotions){
     this.promotions = promotions;
    }
```